syntax = "proto3";
option java_package = "com.aylaasia.referenceapp.grpc";
import "google/protobuf/empty.proto";
import "Base.proto";
/**
设备相关接口
 */
service DeviceService{
  /**
  绑定设备
   */
  rpc bindDevice(BindDeviceReq) returns (Result);

  /**
  设备列表
   */
  rpc getDeviceList(google.protobuf.Empty) returns (DeviceListResp);

  /**
  设备详情
   */
  rpc getDeviceInfo(DeviceReq) returns (Device);

  /**
  设备解绑
   */
  rpc unBindDevice(UnBindDeviceReq) returns (UnBindDeviceResp);

  /**
  设置设备属性值
   */
  rpc setDeviceProperty(SetDevicePropertyReq) returns (Result);

  /**
  获取设备属性值
   */
  rpc getDeviceProperty(GetDevicePropertyReq) returns (DevicePropertyResp);

  /**
  获取设备所有属性值
   */
  rpc getDeviceProperties(GetDevicePropertiesReq) returns (GetDevicePropertiesResp);

  /**
  编辑设备信息
   */
  rpc updateDevice(UpdateDeviceReq) returns (Device);
}

/**
绑定设备入参
 */
message BindDeviceReq{
  /**
    原生iot云类型 0是ayla云，1是阿里云
   */
  int32 cuId = 1;
  /**
  设备id
   */
  string deviceId = 2;
  /**
   设备名称
   */
  string nickName = 3;
}

/**
设备列表出参
 */
message DeviceListResp{
  /**
  设备详情 list
   */
  repeated Device devices = 1;
}

/**
设备详情入参
 */
message DeviceReq{
  /**
  设备id
   */
  string deviceId = 1;
}
/**
设备详情出参
 */
message Device{
  /**
  设备id
   */
  string deviceId = 1;
  /**
  设备名称
   */
  string nickName = 2;
  /**
  连接状态 在线ONLINE 离线OFFLINE
   */
  string connectionStatus = 3;
  /**
  固件版本号
   */
  string firmwareVersion = 4;
  /**
  mac地址
   */
  string macAddress = 5;
  /**
  协议类型
   */
  string connectivity = 6;

}

/**
解绑设备入参
 */
message UnBindDeviceReq{
  /**
  设备id
   */
  string deviceId = 1;
}

/**
解绑设备出参
 */
message UnBindDeviceResp{

  /**
  设备id
   */
  string deviceId = 1;
  /**
  是否成功，true成功，false失败
   */
  bool result = 2;
}

/**
设置设备属性值入参
 */
message SetDevicePropertyReq{
  /**
  设备id
   */
  string deviceId = 1;
  /**
  设备属性
   */
  string propertyName = 2;
  /**
  设置属性值
   */
  string propertyValue = 3;
}

/**
获取设备属性值入参
 */
message GetDevicePropertyReq{
  /**
  设备id
   */
  string deviceId = 1;
  /**
  设备属性值
   */
  string propertyName = 2;
}

/**
获取设备属性值出参
 */
message DevicePropertyResp{

  /**
  设备id
   */
  string deviceId = 1;

  /**
  设备属性
   */
  DeviceProperty deviceProperty = 2;
}

/**
获取设备所有属性值入参
 */
message GetDevicePropertiesReq{
  /**
  设备id
   */
  string deviceId = 1;
}

/**
获取设备所有属性值出参
 */
message GetDevicePropertiesResp{
  /**
  属性值列表
   */
  repeated DeviceProperty deviceProperties = 1;
}

/**
设备属性
 */
message DeviceProperty{
  /**
  设备属性名
   */
  string propertyName = 1;
  /**
  设备属性值
   */
  string propertyValue = 2;
}

/**
编辑设备信息入参
 */
message UpdateDeviceReq{
  /**
  设备id
   */
  string deviceId = 1;
  /**
  设备名称
   */
  string nickName = 2;
}
